server/src/main/java/org/elasticsearch/action/admin/indices/template/put/TransportPutComponentTemplateAction.java

```diff
@@ -91,14 +117,36 @@ protected void masterOperation(
         final ActionListener<AcknowledgedResponse> listener
     ) {
         ComponentTemplate componentTemplate = normalizeComponentTemplate(request.componentTemplate(), indexScopedSettings);
-        indexTemplateService.putComponentTemplate(
-            request.cause(),
-            request.create(),
-            request.name(),
-            request.masterNodeTimeout(),
-            componentTemplate,
-            listener
-        );
+        if (componentTemplate.hasDataLifecycle()) {
+            var composableTemplates = indexTemplateService.getTemplatesUsingComponent(state, request.name());
```

Comment:
Note: in the final implementation, we will likely simulate the index templates and check if the final simulated templates have life-cycles. This will be necessary since it will be possible to let component templates (and the index template) "nullify" a life-cycle, by specifying it as `"lifecycle": null` in the configuration.